packages/flutter/lib/src/widgets/actions.dart

```diff
@@ -251,6 +251,23 @@ abstract class Action<T extends Intent> with Diagnosticable {
   /// The default implementation returns true.
   bool consumesKey(T intent) => true;
 
+  /// Converts the result of [invoke] of this action to a [KeyEventResult].
+  ///
+  /// This is typically used when the action is invoked in response to a keyboard
+  /// shortcut.
+  ///
+  /// By default, calls [consumesKey] and converts the returned boolean to
+  /// [KeyEventResult.handled] if it's true, and [KeyEventResult.skipRemainingHandlers]
+  /// if it's false.
+  ///
+  /// Concrete implementations may refine the type of [actionResult], since
+  /// they know type returned by [invoke].
+  KeyEventResult toKeyEventResult(T intent, covariant Object? actionResult) {
+    return consumesKey(intent)
+      ? KeyEventResult.handled
+      : KeyEventResult.skipRemainingHandlers;
```

Comment:
Maybe actions need to return a KeyEventResult instead of a bool from consumesKey?

Reply:
This conversion logic is [the same as it was before](https://github.com/flutter/flutter/blob/ecf9b2d20a8fce893c1c68d49b0ea39897946432/packages/flutter/lib/src/widgets/shortcuts.dart#L855). I simply moved it from `shortcuts.dart` to this method so that certain action implementations could customize it.

We could reevaluate it, but one critical piece is that as of right now `KeyEventResult.ignored` will not cause the engine to propagate the event to the platform.

Reply:
Hmm, OK, I guess I missed that it was the same before.